Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -180,6 +233,37 @@ public TimeSpan UpdateState(string instrumentationKey)
             return this.DetermineBackOffs();
         }
 
+        private void UpdateConfiguration(CollectionConfigurationInfo configurationInfo)
+        {
+            // we only get here if Etag in the header is different from the current one, but we still want to check if Etag in the body is also different
+            if (configurationInfo != null && !string.Equals(configurationInfo.ETag, this.currentConfigurationETag, StringComparison.Ordinal))
+            {
+                this.collectionConfigurationErrors.Clear();
+
+                CollectionConfigurationError[] errors = null;
+                try
+                {
+                    errors = this.onUpdatedConfiguration?.Invoke(configurationInfo);
+                }
+                catch (Exception e)
+                {
+                    this.collectionConfigurationErrors.Add(
+                        CollectionConfigurationError.CreateError(
+                            CollectionConfigurationErrorType.CollectionConfigurationFailureToCreateUnexpected,
+                            string.Format(CultureInfo.InvariantCulture, "Unexpected error applying configuration. ETag: {0}", configurationInfo.ETag ?? string.Empty),
+                            e,
+                            Tuple.Create("ETag", configurationInfo.ETag)));
+                }
+
+                if (errors != null)
+                {
+                    this.collectionConfigurationErrors.AddRange(errors);
+                }
+
+                this.currentConfigurationETag = configurationInfo.ETag;
+            }
```

Comment:
- the service will send a default configuration if it wants to "reset" to default.
- if something horrible happens and the UX -> QPS pipeline somehow crashes/resets/reboots - the service will send the default configuration and SDK will get to the default state.
- the service doesn't need to remember any etags, it will send new configuration whenever its etag is different from what SDK provides, so any misfortune with service's internal state will lead to SDK getting a default configuration.
We can't stop collection if the service is unreachable since that might be transient. If the service is unreachable for a while - we stop collecting anything anyway, and the configuration is not important.